packages/manager/apps/web/client/app/hosting/multisite/hosting-multisite.service.js

```diff
@@ -161,48 +271,52 @@ angular.module('services').service(
       runtimeId,
       serviceName,
     ) {
-      return this.getZoneLinked(domain)
-        .then((urlSplitted) => {
-          let baseDomain;
-          let domainName;
-
-          if (urlSplitted.zone) {
-            baseDomain = urlSplitted.zone;
-            domainName = urlSplitted.subDomain;
-          } else {
-            baseDomain = domain;
-            domainName = null;
-          }
+      const ipLocation = countryIp?.country ? countryIp.country : null;
+
+      const updateDomain = this.RequestAttachedDomainUpdate(
+        serviceName,
+        domain,
+        cdn.toLowerCase(),
+        firewall.toLowerCase(),
+        ownLog,
+        home,
+        runtimeId,
+        ssl,
+        false,
+        ipLocation,
+      );
 
-          return this.addDomain(
-            baseDomain,
-            domainName,
-            home,
-            wwwNeeded,
-            ipv6Needed,
-            !!urlSplitted.zone,
-            cdn.toLowerCase(),
-            countryIp,
-            firewall.toLowerCase(),
-            ownLog,
-            ssl,
-            runtimeId,
-            serviceName,
-          );
-        })
-        .then((response) => {
-          this.Hosting.resetDomains();
-          this.getTaskIds({ fn: 'attachedDomain/update' }, serviceName).then(
-            (taskIds) => {
-              this.pollRequest({
-                serviceName,
-                taskIds,
-                namespace: 'modifyDomain',
-              });
-            },
-          );
-          return response.data;
-        });
+      const promises = [updateDomain];
+
+      if (wwwNeeded) {
+        const updateDomainwww = this.RequestAttachedDomainUpdate(
+          serviceName,
+          `www.${domain}`,
+          cdn.toLowerCase(),
+          firewall.toLowerCase(),
+          ownLog,
+          home,
+          runtimeId,
+          ssl,
+          false,
+          ipLocation,
+        );
+
+        promises.push(updateDomainwww);
+      }
+
+      return Promise.all(promises).then(() => {
```

Comment:
**nitpick:** Instead of using the native JS Promise, better to use `$q.all` as there change detection issues when native JS Promises are used with AngularJS.